common/kokkos-sampler/kp_sampler_skip.cpp

```diff
@@ -0,0 +1,239 @@
+    profileLibrary = getenv("KOKKOS_PROFILE_LIBRARY");
+    if (NULL == profileLibrary) {
+      printf("KokkosP: No library to call in %s\n", profileLibrary);
+      exit(-1);
+    }
+  }
+
+  char* envBuffer = (char*)malloc(sizeof(char) * (strlen(profileLibrary) + 1));
+  sprintf(envBuffer, "%s", profileLibrary);
+
+  char* nextLibrary = strtok(envBuffer, ";");
+
+  for (int i = 0; i < loadSeq; i++) {
+    nextLibrary = strtok(NULL, ";");
+  }
+
+  nextLibrary = strtok(NULL, ";");
+
+  if (NULL == nextLibrary) {
+    printf("KokkosP: No child library to call in %s\n", profileLibrary);
+    exit(-1);
+  } else {
+    if (tool_verbosity > 0) {
+      printf("KokkosP: Next library to call: %s\n", nextLibrary);
+      printf("KokkosP: Loading child library ..\n");
+    }
+
+    void* childLibrary = dlopen(nextLibrary, RTLD_NOW | RTLD_GLOBAL);
+
+    if (NULL == childLibrary) {
+      fprintf(stderr, "KokkosP: Error: Unable to load: %s (Error=%s)\n",
+              nextLibrary, dlerror());
+      exit(-1);
+    } else {
+      beginForCallee =
+          (beginFunction)dlsym(childLibrary, "kokkosp_begin_parallel_for");
+      beginScanCallee =
+          (beginFunction)dlsym(childLibrary, "kokkosp_begin_parallel_scan");
+      beginReduceCallee =
+          (beginFunction)dlsym(childLibrary, "kokkosp_begin_parallel_reduce");
+
+      endScanCallee =
+          (endFunction)dlsym(childLibrary, "kokkosp_end_parallel_scan");
+      endForCallee =
+          (endFunction)dlsym(childLibrary, "kokkosp_end_parallel_for");
+      endReduceCallee =
+          (endFunction)dlsym(childLibrary, "kokkosp_end_parallel_reduce");
+
+      initProfileLibrary =
+          (initFunction)dlsym(childLibrary, "kokkosp_init_library");
+      finalizeProfileLibrary =
+          (finalizeFunction)dlsym(childLibrary, "kokkosp_finalize_library");
+
+      if (NULL != initProfileLibrary) {
+        (*initProfileLibrary)(loadSeq + 1, interfaceVer, devInfoCount,
+                              deviceInfo);
+      }
+
+      if (tool_verbosity > 0) {
+        printf("KokkosP: Function Status:\n");
+        printf("KokkosP: begin-parallel-for:      %s\n",
+               (beginForCallee == NULL) ? "no" : "yes");
+        printf("KokkosP: begin-parallel-scan:     %s\n",
+               (beginScanCallee == NULL) ? "no" : "yes");
+        printf("KokkosP: begin-parallel-reduce:   %s\n",
+               (beginReduceCallee == NULL) ? "no" : "yes");
+        printf("KokkosP: end-parallel-for:        %s\n",
+               (endForCallee == NULL) ? "no" : "yes");
+        printf("KokkosP: end-parallel-scan:       %s\n",
+               (endScanCallee == NULL) ? "no" : "yes");
+        printf("KokkosP: end-parallel-reduce:     %s\n",
+               (endReduceCallee == NULL) ? "no" : "yes");
+      }
+    }
+  }
+
+  free(envBuffer);
+
+  kernelCounter = 0;
+  uniqID        = 1;
+
+  const char* tool_sample = getenv("KOKKOS_TOOLS_SAMPLE_SKIP");
```

Comment:
this should be `KOKKOS_TOOLS_SAMPLER_SKIP` so we are consistent.

Reply:
I have changed this, also updating the Kokkos Tools wiki name of this tool back to sampler and not sample.